autokeras/tasks/image.py

```diff
@@ -221,3 +221,107 @@ def fit(self,
+class ImageSegmenter(SupervisedImagePipeline):
+    """AutoKeras image segmentation class.
+
+    # Arguments
+        num_classes: Int. Defaults to None. If None, it will infer from the data.
+        multi_label: Boolean. Defaults to False.
+        loss: A Keras loss function. Defaults to use 'binary_crossentropy' or
+            'categorical_crossentropy' based on the number of classes.
+        metrics: A list of metrics used to measure the accuracy of the model,
+            default to 'accuracy'.
+        name: String. The name of the AutoModel. Defaults to 'image_segmenter'.
+        max_trials: Int. The maximum number of different Keras Models to try.
+            The search may finish before reaching the max_trials. Defaults to 100.
+        directory: String. The path to a directory for storing the search outputs.
+            Defaults to None, which would create a folder with the name of the
+            AutoModel in the current directory.
+        objective: String. Name of model metric to minimize
+            or maximize, e.g. 'val_accuracy'. Defaults to 'val_loss'.
+        overwrite: Boolean. Defaults to `True`. If `False`, reloads an existing
+            project of the same name if one is found. Otherwise, overwrites the
+            project.
+        seed: Int. Random seed.
+    """
+
+    def __init__(self,
+                 num_classes=None,
+                 multi_label=False,
+                 loss=None,
+                 metrics=None,
+                 name='image_segmenter',
+                 max_trials=100,
+                 directory=None,
+                 objective='val_accuracy',
+                 overwrite=True,
+                 seed=None):
+        super().__init__(
+            outputs=hypermodels.SegmenterHead(num_classes=num_classes,
+                                              multi_label=multi_label,
+                                              loss=loss,
+                                              metrics=metrics),
+            max_trials=max_trials,
+            directory=directory,
+            name=name,
+            objective=objective,
+            tuner=greedy.Greedy,
+            overwrite=overwrite,
+            seed=seed)
+
+    def fit(self,
+            x=None,
+            y=None,
+            epochs=None,
+            callbacks=None,
+            validation_split=0.2,
+            validation_data=None,
+            **kwargs):
+        """Search for the best model and hyperparameters for the AutoModel.
+
+        It will search for the best model based on the performances on
+        validation data.
+
+        # Arguments
+            x: numpy.ndarray or tensorflow.Dataset. Training image dataset x.
+                The shape of the data should be 3 or 4 dimensional, the last
+                dimension of which should be channel dimension.
+            y: numpy.ndarray or tensorflow.Dataset. Training image data
+                set y. It should be a tensor with the same shape of x. Each
```

Comment:
It's not the same shape since the number of channels will be different. It's the same *height* and *width*.